CRM/Activity/Import/Parser/Activity.php

```diff
@@ -132,9 +132,21 @@ public function summary(&$values) {
     $this->setActiveFieldValues($values);
 
     try {
+      // Check required fields if this is not an update.
+      if (!$this->getFieldValue($values, 'activity_id')) {
+        if (!$this->getFieldValue($values, 'activity_label')
+        && !$this->getFieldValue($values, 'activity_type_id')) {
+          throw new CRM_Core_Exception(ts('Missing required fields: Activity type label or Activity type ID'));
+        }
+        if (!$this->getFieldValue($values, 'activity_date_time')) {
+          throw new CRM_Core_Exception(ts('Missing required fields'));
```

Comment:
@eileenmcnaughton any reason why you split the activity_date_time out from the other 2?

Reply:
The other 2 are an either or so a bit more complex